pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -411,4 +415,69 @@ public static FilterContext getFilter(FilterQueryTree node) {
         throw new IllegalStateException();
     }
   }
+
+  public static FilterContext overrideFilterWithExpressionOverrideHints(FilterContext filter, IndexSegment indexSegment,
```

Comment:
We should move these data dependent methods to a separate util (or put them in `InstancePlanMakerImplV2` for now, and extract them out in the future if needed to be reused). IMO `RequestContextUtils` should be data independent, and only related to query.